### PR TITLE
REDSHOP-5778 [Bug] Fix bug of Wrapper

### PR DESCRIPTION
**Pull request by dungdangvan**

### CODE REVIEW

File: component/admin/models/wrapper.php [hunk 1]
```diff
@@ -12,123 +12,134 @@
...
+
+	public function _buildQuery()
+	{
+		$db = \JFactory::getDbo();
+		$app = \JFactory::getApplication();
+		$showAll = $app->input->get('showall', '0');
+		$subQuery = [];
+
+		if ($showAll && $this->_productid != 0) {
+			$subQuery[] = 'FIND_IN_SET(' . $this->_productid . ',w.product_id) OR wrapper_use_to_all = 1';
```
Comment (anhredweb) [inline component/admin/models/wrapper.php]: @dungdangvan : please adjust:` $db->q($this->_productid)`, `$db->qn('w.product_id')`, `$db->qn('wrapper_use_to_all') = $db->q(1)`

File: component/admin/models/wrapper.php [hunk 2]
```diff
@@ -12,123 +12,134 @@
...
+
+			$query = $db->getQuery(true)
+				->select('*')
+				->from($db->qn('#__redshop_product_category_xref'))
+				->where($db->qn('product_id') . ' = ' . $db->q((int)$this->_productid));
+			$db->setQuery($query);
+			$cat = $db->loadObjectList();
+
+			for ($i = 0, $in = count($cat); $i < $in; $i++) {
+				$subQuery[] = 'FIND_IN_SET(' . $cat[$i]->category_id . ',category_id)';
```
Comment (anhredweb) [inline component/admin/models/wrapper.php]: @dungdangvan same here, please use `$db->qn` and `$db->q`

File: component/admin/models/wrapper.php [hunk 3]
```diff
@@ -12,123 +12,134 @@
...
+
+		if (!empty($subQuery)) {
+			$query->where('(' . implode(' OR ', $subQuery) . ')');
+		}
+
+		$filter = $this->getState('filter');
+		$filter = $db->escape(trim($filter));
+
+		if ($filter) {
+			$query->where($db->qn('w.wrapper_name') . " LIKE '%" . $filter . "%' ");
```
Comment (anhredweb) [inline component/admin/models/wrapper.php]: @dungdangvan `$db->quote($db->escape('%' . $filter . '%'))`

**Review by anhredweb — DISMISSED** [review]

File: component/admin/models/wrapper.php (hunk 3 above)
Comment (dungdangvan) [reply component/admin/models/wrapper.php]: It has already been escaped, please see above code

**Review by anhredweb — DISMISSED** [review]

**Pull request merged**